lib/hooks/queries/useFeePayingAsset.ts

```diff
@@ -115,3 +113,43 @@ export const useFeePayingAsset = (
 
   return query;
 };
+
+const findBestFeePayingAsset = (
+  foreignAssetBalances: CurrencyBalance[],
+  assetMetadata: [number | "Ztg", AssetMetadata][],
+  baseFee: Decimal,
+  constants: ChainConstants,
+) => {
+  // find first available asset to pay fee, else just return native asset
+  const availableAsset = foreignAssetBalances.find((asset) => {
+    const feeFactor = assetMetadata
```

Comment:
I think you can refactor this as a function so you don't have to repeat this later on line 136